pyface/util/font_parser.py

```diff
@@ -0,0 +1,106 @@
+# (C) Copyright 2005-2021 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+
+GENERIC_FAMILIES = {
+    'default', 'fantasy', 'decorative', 'serif', 'roman', 'cursive', 'script',
+    'sans-serif', 'swiss', 'monospace', 'modern', 'typewriter', 'teletype'
+}
+WEIGHTS = {'thin', 'extra-light', 'light', 'regular', 'medium', 'demi-bold',
+    'bold', 'extra-bold', 'heavy', 'extra-heavy'}
+STRETCHES = {'ultra-condensed', 'extra-condensed', 'condensed',
+    'semi-condensed', 'semi-expanded', 'expanded', 'extra-expanded',
+    'ultra-expanded'}
+STYLES = {'italic', 'oblique'}
+VARIANTS = {'small-caps'}
+DECORATIONS = {'underline', 'strikethrough', 'overline'}
+NOISE = {'pt', 'point', 'px', 'family'}
+
```

Comment:
Would it be possible to add an EBNF (or other) description of the grammar being parsed, either here or in the docs somewhere (or both)?

Reply:
It's probably not worth it for this parser: it's more-or-less a "bag of words" designed to be backwards compatible with the other ad-hoc parsers in TraitsUI and Enable.  It _is_ probably a good idea to have the "words" documented somewhere.

Longer-term I do want a proper parser with a well-defined grammar, which is why the parser is designed to be able to be switched out.

Reply:
Okay, then maybe just an informal description in the comments here? Otherwise there's no description of the intended behaviour outside the code itself.

Reply:
I've expanded the parser docstring accordingly.